eng/pipelines/templates/steps/npm-release-task.yml

```diff
@@ -10,6 +9,22 @@ steps:
   - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
 
   - ${{ if eq(parameters.Registry, 'https://registry.npmjs.org/') }}:
+    - pwsh: |
+        set -o xtrace
+        $tarFile = (Get-ChildItem -Path "${{parameters.Artifact.name}}*.tgz")?.Name
+        $tempDir = "temp_decompress"
+        New-Item -ItemType Directory -Force -Path $tempDir
+        tar -xzf $tarFile -C $tempDir
+        $pkg = Get-Content -Raw "$tempDir\package\package.json" | ConvertFrom-Json
+        $packageName = $pkg.Name
+        $packageVersion = $pkg.Version
+        $packageProps = npm view $packageName -json | ConvertFrom-Json
+        $originalTags = $packageProps.'dist-tags' | ConvertTo-Json -Compress
+        echo "##vso[task.setvariable variable=OriginalTags]$originalTags
+        echo "##vso[task.setvariable variable=IntendedTagVersion]$packageVersion
+        Remove-Item -Force -Recurse $tempDir
```

Comment:
Do we need to delete? Can we share with the later steps? Perhaps another variable with the location of the unpacked package?